### DISCUSSION TITLE
How to write .projenrc.js as ES module

### DISCUSSION THREAD

**Discussion by jshinevar** [projen/projen#2803] [Q&A]
When I run `npx projen` on a new cdk construct library, I am getting the following error:

```
❯ npx projen
👾 default | node .projenrc.js
(node:96407) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/me/Documents/my-project/.projenrc.js:1
import { awscdk } from 'projen';
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47

Node.js v18.16.1
👾 Synthesis failed: Task "default" failed when executing "node .projenrc.js"
```

I can temporarily solve this by adding `"type": "module"` to the package.json and the first time I run `npx projen` it works, but that is until it generates a new package.json file.  How can I add that to the `.projenrc.js` so that it's a part of the package.json every time?

**Comment by mrgrain** [comment]
Try this:
```ts
project.addField('type', 'module');
```

**Comment by mrgrain** [comment]
Or if you don't actually want to use ES modules, re-write the `import` statement to use `require()`

Or use TypeScript instead. Here's how you can migrate your rcfile: 
https://projen.io/docs/quick-starts/typescript/hello-world/#migrating-projenrcjs-to-projenrcts

**Reply by jshinevar** [reply]
I would much rather use TS, I'll give this a shot shortly, thank you.

**Reply by tmokmss** [reply]
fyi: here's the updated link https://projen.io/docs/quick-starts/typescript/hello-world/#migrating-projenrcjs-to-projenrcts
